### ISSUE TITLE
placeholder attribute announced with Safari + VO

### ISSUE THREAD

**Issue by scottaohara** [accessabilly/accessabilly-upload#1]
Hey @accessabilly 

Was giving your script a run through the other day. I noticed that when a file has been selected, the text input that displays the file name, announces the following in voice over:

"file-name, uploaded file(label), text, no file uploaded"

The following is what a native file input announces in Safari + VO
"file-name, Label, file upload button"

Being that you're recreating the file input so it can be styled, rather than relying on the native file input, it's understandable that the announcements wouldn't be exactly the same (hence the additional announcement of the text input).  But for VoiceOver in particular this announcement is confusing due to the fact VO will announce the placeholder attribute even after a value has been passed into the text input.

To mitigate this, it'd be best to remove the placeholder value (or attribute all together) when a file has been selected, so that the "no file uploaded" isn't announced.

**Comment by accessabilly** [comment]
Hi @scottaohara 
Sorry for my late reply, I was busy around the new year. Mea culpa.
Thank your for your feedback. I must admit that I haven't tested this stuff in VocieOver yet.

I checked that again in JAWS and NVDA on Windows and it doesn't seem to be harmful there. The placeholder was only spoken initially. 

To fix this issue instead of removing the placeholder completely I decided to change its contents to the filename. So when you upload a file, its name will be in the value and in the palceholder as well.

Hope this fixes it for you, too.

Cheers from Switzerland!
